legal-api/tests/unit/resources/v2/test_business.py

```diff
@@ -338,6 +338,48 @@ def test_get_business_with_court_orders(session, client, jwt):
     assert rv.json['business']['identifier'] == identifier
     assert rv.json['business']['hasCourtOrders'] == True
 
+@pytest.mark.parametrize('identifier, legal_type, draft_type, nr_number', [
+    ('Tb31yQIuBw', 'COMP', 'ATMP', None),
+    # Add more scenarios here as needed
+])
+def test_draft_amalgamation_name_selection(session, client, jwt, identifier, legal_type, draft_type, nr_number):
+    """Test draft regular amalgamation with various name selection scenarios."""
+    # Setup a temporary registration and draft filing
+    temp_reg = RegistrationBootstrap()
+    temp_reg._identifier = identifier
+    temp_reg.save()
+
+    json_data = copy.deepcopy(FILING_HEADER)
+    json_data['filing']['header'] = {
+        'name': 'amalgamationApplication',
+        'identifier': identifier,
+        'legalType': legal_type,
+    }
+    json_data['filing']['amalgamationApplication'] = {
+        'type': 'regular',
+        'nameRequest': {
+            'nrNumber': nr_number,
```

Comment:
legalType is mandatory in nameRequest object (https://github.com/bcgov/business-schemas/blob/main/src/registry_schemas/schemas/name_request.json#L53)